engine/src/cmd/movable.cpp

```diff
@@ -0,0 +1,432 @@
+    float  lastWarpField = graphicOptions.WarpFieldStrength;
+
+    bool   playa = isPlayerShip();
+
+    float  warprampuptime = playa ? humanwarprampuptime : compwarprampuptime;
+    //Warp Turning on/off
+    if (graphicOptions.WarpRamping) {
+        float  oldrampcounter = graphicOptions.RampCounter;
+        if (graphicOptions.InWarp == 1)             //Warp Turning on
+            graphicOptions.RampCounter = warprampuptime;
+        else                                        //Warp Turning off
+            graphicOptions.RampCounter = warprampdowntime;
+        //switched mid - ramp time; we also know old mode's ramptime != 0, or there won't be ramping
+        if (oldrampcounter != 0 && graphicOptions.RampCounter != 0 ) {
+            if (graphicOptions.InWarp == 1)             //Warp is turning on before it turned off
+                graphicOptions.RampCounter *= (1 - oldrampcounter/warprampdowntime);
+            else                                        //Warp is turning off before it turned on
+                graphicOptions.RampCounter *= (1 - oldrampcounter/warprampuptime);
+        }
+        graphicOptions.WarpRamping = 0;
+    }
+    if (graphicOptions.InWarp == 1 || graphicOptions.RampCounter != 0) {
+        float rampmult = 1.f;
+        if (graphicOptions.RampCounter != 0) {
+            graphicOptions.RampCounter -= SIMULATION_ATOM;
+            if (graphicOptions.RampCounter <= 0)
+                graphicOptions.RampCounter = 0;
+            if (graphicOptions.InWarp == 0 && graphicOptions.RampCounter > warprampdowntime)
+                graphicOptions.RampCounter = (1-graphicOptions.RampCounter/warprampuptime)*warprampdowntime;
+            if (graphicOptions.InWarp == 1 && graphicOptions.RampCounter > warprampuptime)
+                graphicOptions.RampCounter = warprampuptime;
+            rampmult = (graphicOptions.InWarp) ? 1.0
+                        -( (graphicOptions.RampCounter
+                            /warprampuptime)
+                            *(graphicOptions.RampCounter
+                            /warprampuptime) ) : (graphicOptions.RampCounter
+                                                    /warprampdowntime)*(graphicOptions.RampCounter/warprampdowntime);
+        }
+        graphicOptions.WarpFieldStrength = GetMaxWarpFieldStrength(rampmult);
+    } else {
+        graphicOptions.WarpFieldStrength = 1;
+    }
+    //not any more? lastWarpField=1;
+    Vector v;
+    if (graphicOptions.WarpFieldStrength != 1.0)
+        v = GetWarpVelocity();
+    else
+        v = Velocity;
+    static float WARPMEMORYEFFECT = XMLSupport::parse_float( vs_config->getVariable( "physics", "WarpMemoryEffect", "0.9" ) );
+    graphicOptions.WarpFieldStrength = lastWarpField*WARPMEMORYEFFECT+(1.0-WARPMEMORYEFFECT)*graphicOptions.WarpFieldStrength;
+    curr_physical_state.position     = curr_physical_state.position+(v*SIMULATION_ATOM*difficulty).Cast();
```

Comment:
`simulation_atom_var` in place of `SIMULATION_ATOM`

Reply:
Fixed.